xtask/src/sovereign.rs

```diff
@@ -0,0 +1,65 @@
+use duct::{cmd, IntoExecutablePath};
+
+pub struct Sovereign(duct::Handle);
+
+impl Sovereign {
+    // The 'out' argument lets you decide whether to
+    // show or not the standard output of the command
+    pub fn try_new(out: bool) -> anyhow::Result<Self> {
+        // clean rollup db
+        let _ = cmd!("make", "clean")
+            .dir("demo/sovereign/demo-rollup/")
+            .run();
+
+        // Duct does a weird thing with dir:
+        // https://docs.rs/duct/latest/duct/struct.Expression.html#method.dir
+        // and the following does not work
+        // let sovereig_rollup_handle = cmd!("../target/release/sov-demo-rollup")
+        //    .dir("demo/sovereign/demo-rollup/")
+        //    .run()?;
+
+        // run sovereign adapter
+        println!("Launching sovereign rollup");
+        let mut sovereign_cmd = cmd!(
+            "sh",
+            "-c",
+            "cd demo/sovereign/demo-rollup && ./../target/release/sov-demo-rollup",
+        );
+
+        if !out {
+            sovereign_cmd = sovereign_cmd.stdout_null();
+        }
+
+        Ok(Self(sovereign_cmd.start()?))
+    }
+
+    // all the network must be up (relaychain and sugondat-node) plus the
+    // sovereing rollup
+    pub fn test_sovereign_rollup(&self) -> anyhow::Result<()> {
+        println!("run sovereign rollup test");
+
+        cmd!(
+            "sh",
+            "-c",
+            "cd demo/sovereign/demo-rollup && \
+             sleep 10 && \
+             ../target/debug/sov-cli rpc set-url http://127.0.0.1:12345 && \
+             ../target/debug/sov-cli keys import --nickname token_deployer --path ../test-data/keys/token_deployer_private_key.json && \
+             ../target/debug/sov-cli transactions import from-file bank --path ../test-data/requests/create_token.json && \
+             ../target/debug/sov-cli transactions import from-file bank --path ../test-data/requests/mint.json && \
+             ../target/debug/sov-cli rpc submit-batch by-nickname token_deployer && \
+             sleep 30 && \
+             echo \"4000 tokens should be minted\" && \
+             curl -X POST -H \"Content-Type: application/json\" -d '{\"jsonrpc\":\"2.0\",\"method\":\"bank_supplyOf\",\"params\":[\"sov1zdwj8thgev2u3yyrrlekmvtsz4av4tp3m7dm5mx5peejnesga27svq9m72\"],\"id\":1}' http://127.0.0.1:12345",
+        ).run()?;
```

Comment:
yeah, or at least a separate file which is `include!`d into the source

Reply:
That was only a quick experiment to see if what I was thinking was possible. It is, so I will find a cleaner way to do it 👍🏻  I was thinking of not using `sh` at all and just running the tasks with `cmd!`

Reply:
What I've implemented in the new commit is just an experiment. I have never written a "shell" script in Rust, so it was worth a try.

If you don't like it, it would be very simple to just execute the .sh file